peer.go

```diff
@@ -26,30 +38,80 @@ type Modify struct {
 // peer is the default implementation of the Peer interface,
 // and it cannot expose to user.
 type peer struct {
-	coor Coordinator
+	coor   Coordinator
+	zkConn *zk.Conn
 }
 
-func StartPeer(c *Config) *peer {
+func StartPeer(c *Config) (*peer, error) {
 	p := &peer{}
 	p.coor = newCoor(c)
-	// TODO: create a zk client and run
-	return p
+	// create a zk client and run
+	// TODO: implement function callback
+	eventCallback := zk.WithEventCallback(callback)
+	var err error
+	p.zkConn, _, err = zk.Connect(c.ZKAddrs, ZkConnTimeOut, eventCallback)
+	if err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+// TODO:Handling various callbacks
+func callback(event zk.Event) {
+	if event.Type == zk.EventSession {
+		return
+	}
+	// TODO: refining this function
 }
 
 func (p *peer) Propose(m *Modify) error {
 	//key, value := m.Key, m.Value
 	// TODO: get version in zk to judge if our proposal is stale
 	switch m.CmdType {
 	case PutCommand:
-		// TODO: Send data to zk
+		// TODO: handling errors
+		p.putToZk(m)
 	case DelCommand:
-		// TODO: Send data to zk
+		// TODO: handling errors
+		p.delToZk(m)
 	default:
-		return errors.New("CommandTypeError: We support only Put command and Delete command")
+		return ErrCommandType
 	}
 	return nil
 }
 
+func (p *peer) putToZk(m *Modify) error {
```

Comment:
In this function, we just need `m.Key` and `m.Value`. Similarly, in the `delToZk()`, we only need `m.Key`. So there is no need to pass the whole struct.